swap/swap.go

```diff
@@ -225,6 +188,9 @@ var defaultCashCheque = cashCheque
 // handleEmitChequeMsg should be handled by the creditor when it receives
 // a cheque from a debitor
 func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitChequeMsg) error {
```

Comment:
Even this whole implementation could be moved to `Peer`. But I think it makes sense to keep `processAndVerifyCheque` on `Swap`, also we have been requested to have all message handlers on `Swap`, so we probably want to keep it exactly the way you have it here.

Reply:
I also think it could be moved. I left it on Swap because of the incentives PR comment.

Reply:
Voting for moving it as well.

Reply:
Seems like unnecessary indirection. In that case the handler should just be directly on `Peer`. Tbh I never understood the argument for moving this to `Swap` in the first place.

Reply:
it seems indeed this function should be on the peer

Reply:
This used to be on `Peer` in the past but all the handle functions where moved as a result of this comment (https://github.com/ethersphere/swarm/pull/1554#discussion_r311536234). But perhaps this was meant only as `handleMsg ` should be on `Peer`. Either way I'm merging this PR as it is now, we can change this in a subsequent one if we want.